yt/frontends/boxlib/data_structures.py

```diff
@@ -733,7 +733,7 @@ def _parse_cparams(self):
             elif param == "amr.ref_ratio":
                 vals = self.refine_by = int(vals[0])
             elif param == "Prob.lo_bc":
-                vals = self.periodicity = ensure_tuple([i == 0 for i in vals.split()])
+                vals = self.periodicity = ensure_tuple([p == "1" for p in vals.split()])
```

Comment:
this line is clearly bugged since it relies on comparing a string to an integer. From what I think I understood, this should be the correct version for it.